lua/cmp_path/init.lua

```diff
@@ -132,62 +118,46 @@ source._candidates = function(_, params, dirname, offset, callback)
       break
     end
 
-    local accept = false
-    accept = accept or include_hidden
-    accept = accept or name:sub(1, 1) ~= '.'
+    if not (include_hidden or string.sub(name, 1, 1) ~= '.') then
+      goto continue
+    end
 
-    local stat = nil
-    -- Stat when fs_scandir_next doesn't return file type
-    if type == nil then
-      stat = vim.loop.fs_stat(dirname .. '/' .. name)
-      if not stat then
-        break
-      end
+    local path = dirname .. '/' .. name
+    local stat = vim.loop.fs_stat(path)
```

Comment:
This change causes a call to fs_stat for every file. Is this intentional?

Reply:
Yes. The logic for calling `fs_stat` lazily was getting too complex, so I decided to always stat every file right away. This makes the `_candidates` function about four times slower: using my `/usr/bin` as a benchmark (4841 entries), without `fs_stat` it takes ~10 milliseconds to run, with - ~40 milliseconds. On my home directory (130 entries) it is about 3 times slower: ~0.3 ms without `fs_stat`, ~1 ms with. If this slowdown is undesirable, I can change it back to be lazily evaluated.